test/operators.jl

```diff
@@ -325,7 +325,7 @@ end
     @test Returns(val)(1) === val
     @test sprint(show, Returns(1.0)) == "Returns{Float64}(1.0)"
 
-    illtype = Vector{Core._typevar(:T, Union{}, Any)}
+    illtype = Vector{Core.TypeVar(:T)}
```

Comment:
What goes wrong here?

Reply:
This change was just to use a cleaner `TypeVar` interface in this test, but the problem here was doing our special inlining with `compute_sparams` targeting a `Type` from the arguments, which created some problems with ambiguities. I disabled this kind of inlining if there is an argument `x` such that `x === Type && isdefined(x, :var)`